From c0c403744b1bf4a9790a8fcaabcd60970cbefe06 Mon Sep 17 00:00:00 2001
From: Michael R Sweet <michael.r.sweet@gmail.com>
Date: Tue, 7 Jun 2022 13:45:29 -0400
Subject: [PATCH] Fix OpenSSL crash bug - "tls" pointer wasn't cleared after
 freeing it (Issue #409)

---
 CHANGES.md         | 1 +
 cups/tls-openssl.c | 2 ++
 2 files changed, 3 insertions(+)

#diff --git a/CHANGES.md b/CHANGES.md
#index 8b78b003fa..befbf3ab44 100644
#--- a/CHANGES.md
#+++ b/CHANGES.md
#@@ -6,6 +6,7 @@ Changes in CUPS v2.4.3 (TBA)
# 
# - Added a title with device uri for found network printers (Issues #402, #393)
# - Fixed configuration on RISC-V machines (Issue #404)
#+- Fixed an OpenSSL crash bug (Issue #409)
# 
# 
# Changes in CUPS v2.4.2 (26th May 2022)
diff --git a/cups/tls-openssl.c b/cups/tls-openssl.c
index c3e57742e8..6db9f8a9c2 100644
--- a/cups/tls-openssl.c
+++ b/cups/tls-openssl.c
@@ -1152,6 +1152,8 @@ _httpTLSStop(http_t *http)		// I - Connection to server
   SSL_shutdown(http->tls);
   SSL_CTX_free(context);
   SSL_free(http->tls);
+
+  http->tls = NULL;
 }
 
 
